pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -719,6 +722,24 @@ func (nc *Controller) monitorNodeHealth() error {
 			}
 			if observedReadyCondition.Status == v1.ConditionUnknown {
 				if nc.useTaintBasedEvictions {
+					// If useTaintBasedEvictions is true, we would not evict pods
```

Comment:
if useTaintBasedEvictions is true, does that mean we would never evict pod in any condition other than when they are violating taints ?

Reply:
I think is TaintBasedEvictions feature gate is only used by node lifecycle controller. It would not influence other case that pod would be evicted. For example, some pods would be evicted when node is in stress.